libs/gh/pull_request.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gh
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// ChangelogPR contains all the data necessary for a changelog from the PR
+type ChangelogPR struct {
+	Body   string `json:"body,omitempty"`
+	Number int    `json:"number,omitempty"`
+	Title  string `json:"title,omitempty"`
+	URL    string `json:"url,omitempty"`
+}
+
+// ListChangelogPullRequestsOpts contains options for searching for changelog pull requests.
+type ListChangelogPullRequestsOpts struct {
+	Branch   string
+	FromDate string
+	ToDate   string
+}
+
+// ListChangelogPullRequests will search for pull requests that provide changelog information.
+func ListChangelogPullRequests(ctx context.Context, dir string, opts *ListChangelogPullRequestsOpts) ([]ChangelogPR, error) {
+	var prs []ChangelogPR
+	query := fmt.Sprintf("base:%s merged:%s -label:no-changelog", opts.Branch, dateRangeFormat(opts.FromDate, opts.ToDate))
+
+	data, err := RunCmd(dir, "pr", "list", "--search", query, "--limit", "200", "--json", "number,url,title,body")
```

Comment:
Since we're moving this to shared-workflows, we should use github package we use in other tools here rather than calling CLI: https://github.com/gravitational/shared-workflows/blob/main/bot/internal/github/github.go

Reply:
Something to mention is that packages cannot import `internal` unless they're nested in the same parent directory as `internal`. I'd imagine that everything under the `internal` package would be moved up under `libs`.

Reply:
Yea it'd need to move to a new module under `/libs`.